pkg/csi/service/wcp/controller.go

```diff
@@ -462,18 +462,27 @@ func (c *controller) createBlockVolume(ctx context.Context, req *csi.CreateVolum
 	topologyRequirement = req.GetAccessibilityRequirements()
 	filterSuspendedDatastores := commonco.ContainerOrchestratorUtility.IsFSSEnabled(ctx, common.CnsMgrSuspendCreateVolume)
 	isTKGSHAEnabled := commonco.ContainerOrchestratorUtility.IsFSSEnabled(ctx, common.TKGsHA)
+	isVdppOnStretchedSVEnabled := commonco.ContainerOrchestratorUtility.IsFSSEnabled(ctx, common.VdppOnStretchedSupervisor)
 	topoSegToDatastoresMap := make(map[string][]*cnsvsphere.DatastoreInfo)
 	if isTKGSHAEnabled {
 		// TKGS-HA feature is enabled
-		// Identify the topology keys in Accessibility requirements.
+		// Identify the topology keys in Accessibility requirements and infer the environment type based on these keys.
+
+		// If both zone and hostname labels are present in the request -
+		// it's host local volume provisioning on a stretched supervisor.
+
+		// If only zone label is present in the request -
+		// it's non-host local volume provisioning on a stretched supervisor.
+
+		// If only hostname label is present in the request -
+		// it's host local volume provisioning on a non-stretched supervisor.
+
+		// If neither zone nor hostname label is present in the request -
+		// it's non-host local volume provisioning on a non-stretched supervisor.
 		hostnameLabelPresent, zoneLabelPresent = checkTopologyKeysFromAccessibilityReqs(topologyRequirement)
-		// TODO: TKGS-HA: This case will only arise when spherelet will add zone and hostname labels to CSINodes.
-		// Currently spherelet only accepts hostname. We will handle this case later.
-		if zoneLabelPresent && hostnameLabelPresent {
-			return nil, csifault.CSIUnimplementedFault, logger.LogNewErrorCodef(log, codes.Unimplemented,
-				"support for topology requirement with both zone and hostname labels is not yet implemented.")
+		if isVdppOnStretchedSVEnabled && hostnameLabelPresent {
+			log.Infof("Host Local volume provisioning on stretched supervisor with requirement: %+v", topologyRequirement)
```

Comment:
isVdppOnStretchedSVEnabled will be enabled even on single cluster supervisor. Message should not say `provisioning on stretched supervisor`

Reply:
+1

Reply:
Makes sense.